fendermint/eth/api/src/cache.rs

```diff
@@ -107,4 +110,54 @@ where
         let mut c = self.id_to_addr.lock().unwrap();
         c.insert(id, addr);
     }
+
+    pub fn set_actor_type_for_addr(&self, addr: Address, actor_type: ActorType) {
+        let mut c = self.addr_to_actor_type.lock().unwrap();
+        c.insert(addr, actor_type);
+    }
+
+    pub fn get_actor_type_from_addr(&self, addr: &Address) -> Option<ActorType> {
+        let mut c = self.addr_to_actor_type.lock().unwrap();
+        c.get(addr).cloned()
+    }
+}
+
+#[cfg(test)]
+mod tests {
+    use crate::cache::AddressCache;
+    use crate::state::ActorType;
+    use cid::Cid;
+    use fendermint_rpc::FendermintClient;
+    use fvm_shared::address::Address;
+    use std::str::FromStr;
+    use tendermint_rpc::HttpClient;
+
+    #[test]
+    fn test_read_and_write_addr_to_actor_type() {
+        let client = FendermintClient::new(HttpClient::new("http://localhost").unwrap());
```

Comment:
There is a `MockClient` for them, e.g. [here](https://github.com/consensus-shipyard/ipc/blob/c3e4c020d52decfbc57eb6e2618c947f3f7b29a7/fendermint/vm/snapshot/src/manager.rs#L483).